src/draw/DefaultDrawer.js

```diff
@@ -61,495 +102,484 @@ class DefaultDrawer {
-   * @return {Boolean} Returns true if normally execute.
+   * Convert svg coordinates into screen coordinates
+   * @param {Number} svgX - SVG x coordinate.
+   * @param {Number} svgY - SVG y coordinate.
+   * @param {SVGSVGElement} [svg=null] - SVG referential.
+   * @return {DOMPoint} - The transformed coordinates.
    */
-  drawLinks(links) {
-    if (!links) return;
-
-    if (document.querySelector('.links') === null) {
-      this.d3.select(`#${this.rootId}`)
-        .append('g')
-        .classed('links', true);
-    }
-
-    const linkGen = this.d3.linkHorizontal()
-      .source((data) => this.getAnchorPosition(data).source)
-      .target((data) => this.getAnchorPosition(data).target);
-
-    this.d3.select('.links')
-      .selectAll('.link')
-      .data(links)
-      .join('path')
-      .classed('link', true)
-      .attr('d', linkGen)
-      .attr('fill', 'none')
-      .attr('stroke', 'black')
-      .attr('stroke-width', 2);
+  SVGToScreen(svgX, svgY, svg = null) {
+    svg = svg ?? this.svg.node();
+    const pivotPoint = svg.createSVGPoint();
+    pivotPoint.x = svgX;
+    pivotPoint.y = svgY;
+    return pivotPoint.matrixTransform(svg.getScreenCTM());
   }
 
   /**
-   * Draws all Components in the parentId Element.
-   * @param {Component[]} components - List of components you want to draw.
-   * @param {String} parentId - Id of the container where you want to draw.
-   * @param {ComponentLink[]} links - List of links you want to draw.
-   * @return {Boolean} Returns true if normally execute.
+   * Find a component in the given components tree.
+   * @param {Component} rootNode - The starting point of the search.
+   * @param {String} componentId - The id to search for.
+   * @return {null|Component} - The component or null if not found
    */
-  drawComponents(components, parentId, links, linksDefinition) {
-    const d3Container = this.d3.select(parentId === this.rootId ? `#${parentId}` : `#${parentId} .component-container`)
-      .selectAll(`.${parentId}.component`)
-      .data(components, (data) => data.id);
-
-    const d3Elements = d3Container
-      .enter().append('svg');
-
-    d3Elements
-      .attr('id', (data) => data.id)
-      .attr('class', (data) => `${parentId} ${data.definition.type} component component-${data.definition.model} ${data.definition.isContainer ? 'isContainer' : ''}`)
-      .html((data) => this.resources.models[data.definition.model]);
-
-    this.__drawModels(components, parentId);
+  findComponentById(rootNode, componentId) {
+    let result = null;
+    if (rootNode) {
+      if (rootNode.id === componentId) result = rootNode;
 
-    components.forEach((component) => {
-      this.setSelectionAction(component, components, links);
-      if (component.children.length > 0) {
-        this.drawComponents(component.children, component.id, links, linksDefinition);
+      if (!result && rootNode.children) {
+        result = rootNode.children
```

Comment:
better use a reduce or a for to retrieve only one component and avoid some call to `findComponentById`

Reply:
Updated to use PluginData's method